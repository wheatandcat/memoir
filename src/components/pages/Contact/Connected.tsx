import React, { memo, useCallback, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import TemplateContact from 'components/templates/Contact/Page';
import { useRecoilValue } from 'recoil';
import { userState } from 'store/atoms';
import * as Device from 'expo-device';
import { Platform, Alert, TouchableOpacity } from 'react-native';
import Constants from 'expo-constants';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';

type Props = {};

const url = Constants.expoConfig?.extra?.INQUIRY_API || '';

export type ConnectedType = {
  userID: string;
  loading: boolean;
  send: boolean;
  text: string;
  onContact: (text: string) => void;
  onChangeText: (text: string) => void;
  onClose: () => void;
};

type Request = {
  body: string;
  name: string;
  email: string;
  env: string;
  userID: string;
  device: string;
  category: string;
};

const Connected: React.FC<Props> = () => {
  const user = useRecoilValue(userState);
  const navigation = useNavigation();
  const [loading, setLoading] = useState(false);
  const [send, setSend] = useState(false);
  const [text, setText] = useState('');

  const onContact = useCallback(async () => {
    if (text.length === 0) {
      Alert.alert('コメントが入力されていません');
      return;
    }

    if (loading) {
      return;
    }

    setLoading(true);

    const req: Request = {
      body: text,
      name: user.displayName || '無名',
      email: 'foobazbar@gmail.com',
      userID: user.userID || '無し',
      env: 'アプリ',
      device: `${Platform.OS}/${Platform.Version}/${Device.modelName}/${String(
        Device.osInternalBuildId
      )}`,
      category: 'フィードバックを送信',
    };

    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(req),
    });

    if (!response.ok) {
      setLoading(false);
      Alert.alert('送信に失敗しました');
      return;
    }

    setSend(true);
  }, [user, text, loading]);

  const onClose = useCallback(() => {
    setSend(false);
    navigation.goBack();
  }, [navigation]);

  React.useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: () => (
        <View pr={2} mr={1}>
          <TouchableOpacity onPress={onContact}>
            <Text>送信</Text>
          </TouchableOpacity>
        </View>
      ),
    });
  }, [navigation, onContact]);

  return (
    <TemplateContact
      send={send}
      userID={user.id ?? ''}
      loading={loading}
      onContact={onContact}
      text={text}
      onChangeText={setText}
      onClose={onClose}
    />
  );
};

export default memo(Connected);
