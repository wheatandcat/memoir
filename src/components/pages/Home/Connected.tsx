import React, { memo, useCallback, useState } from 'react';
import { useRecoilState, useRecoilValue } from 'recoil';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { useCreateItemMutation, NewItem } from 'queries/api/index';
import { homeDateState, homeItemsState, Item } from 'store/atoms';
import useHomeItems from 'hooks/useHomeItems';
import { Props as IndexProps } from './';
import Plain from './Plain';

dayjs.locale('ja');
dayjs.extend(advancedFormat);

type Props = IndexProps & {
  openSettingModal: boolean;
  onCloseSettingModal: () => void;
};

type State = {
  openAddItemModal: boolean;
};

export type ConnectedType = {
  items: Item[];
  addItemLoading: boolean;
  date: string;
  openAddItemModal: boolean;
  openSettingModal: boolean;
  onAddItem: (item: NewItem) => void;
  onChangeDate: (date: string) => void;
  onCloseAddItem: () => void;
  onCloseSettingModal: () => void;
  onItem: (itemID: string) => void;
  onMemoir: () => void;
  onOpenAddItem: () => void;
};

const initialState = (): State => ({
  openAddItemModal: false,
});

const Connected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());
  const [homeDate, setHomeDate] = useRecoilState(homeDateState);
  const homeItems = useRecoilValue(homeItemsState);
  const { loading, error, refetch } = useHomeItems();

  const onOpenAddItem = useCallback(() => {
    setState((s) => ({ ...s, openAddItemModal: true }));
  }, []);

  const onCloseAddItem = useCallback(() => {
    setState((s) => ({ ...s, openAddItemModal: false }));
  }, []);

  const [createItemMutation, createItemMutationData] = useCreateItemMutation({
    async onCompleted() {
      await refetch?.();

      onCloseAddItem();
    },
  });

  const onAddItem = useCallback(
    (newItem: NewItem) => {
      const variables = {
        input: newItem,
      };

      createItemMutation({ variables });
    },
    [createItemMutation]
  );

  const onChangeDate = useCallback(
    (date: string) => {
      setHomeDate({
        date: dayjs(date).format('YYYY-MM-DDT00:00:00+09:00'),
      });
    },
    [setHomeDate]
  );

  const onItem = useCallback(
    (itemID: string) => {
      props.navigation.navigate('ItemDetail', {
        id: itemID,
        date: homeDate.date,
      });
    },
    [props.navigation, homeDate.date]
  );

  const onMemoir = useCallback(() => {
    props.navigation.navigate('Memoir', {
      startDate: dayjs().add(-7, 'day').format('YYYY-MM-DDT00:00:00+09:00'),
      endDate: dayjs().format('YYYY-MM-DDT00:00:00+09:00'),
    });
  }, [props.navigation]);

  return (
    <Plain
      items={homeItems.items}
      loading={loading}
      error={error}
      addItemLoading={createItemMutationData.loading}
      date={dayjs(homeDate.date).format('YYYY-MM-DD')}
      openAddItemModal={state.openAddItemModal}
      openSettingModal={props.openSettingModal}
      onAddItem={onAddItem}
      onChangeDate={onChangeDate}
      onCloseAddItem={onCloseAddItem}
      onCloseSettingModal={props.onCloseSettingModal}
      onItem={onItem}
      onMemoir={onMemoir}
      onOpenAddItem={onOpenAddItem}
    />
  );
};

export default memo(Connected);
