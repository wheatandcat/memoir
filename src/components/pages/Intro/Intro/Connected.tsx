import React, { memo, useCallback, useState } from 'react';
import * as Notifications from 'expo-notifications';
import useMemoirNotificationSetting from 'hooks/useMemoirNotificationSetting';
import { useNotification } from 'containers/Notification';
import Plain from './Plain';

export type Props = {
  onFinish: () => void;
};

export type ConnectedType = {
  step: number;
  onSaveNotification: (input: Input, callback: () => void) => void;
  onStep: (num: number) => void;
};

type Input = {
  dayOfWeek: number;
  hours: number;
  minutes: number;
  notification: boolean;
};

const Connected: React.FC<Props> = () => {
  const memoirNotificationSetting = useMemoirNotificationSetting();
  const { onPermissionRequest } = useNotification();

  const [step, setStep] = useState(0);

  const onStep = useCallback((num: number) => {
    setStep(num);
  }, []);

  const onSaveNotification = useCallback(
    async (input: Input, callback: () => void) => {
      if (input.notification) {
        if (!onPermissionRequest) {
          return;
        }

        onPermissionRequest(() => {
          Notifications.scheduleNotificationAsync({
            content: {
              body: 'ふりかえりの時間になりました',
              data: {
                urlScheme: 'Memoir',
              },
            },
            trigger: {
              hour: input.hours,
              minute: input.minutes,
              weekday: input.dayOfWeek,
              repeats: true,
            },
          });
        });

        // NOTE: API出力
      } else {
        // 通知OFFの場合は、通知をキャンセルする
        Notifications.cancelAllScheduledNotificationsAsync();
      }

      memoirNotificationSetting.onSave(input);
      callback();
    },
    [memoirNotificationSetting, onPermissionRequest]
  );

  return (
    <Plain
      loading={memoirNotificationSetting.loading}
      data={{
        dayOfWeek: memoirNotificationSetting.dayOfWeek,
        hours: memoirNotificationSetting.hours,
        minutes: memoirNotificationSetting.minutes,
        notification: memoirNotificationSetting.notification,
      }}
      onSaveNotification={onSaveNotification}
      step={step}
      onStep={onStep}
    />
  );
};

export default memo(Connected);
