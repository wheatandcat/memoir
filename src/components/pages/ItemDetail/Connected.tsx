import React, { memo, useCallback, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import { useSetRecoilState } from 'recoil';
import dayjs from 'lib/dayjs';
import {
  ItemDocument,
  UpdateItemDocument,
  DeleteItemDocument,
  UpdateItem,
  NewItem,
  DeleteItem,
} from 'queries/api/index';
import { homeDateState } from 'store/atoms';
import useHomeItems from 'hooks/useHomeItems';
import { useQuery, useMutation } from '@apollo/client';
import Plain from './Plain';

export type Props = {
  itemID: string;
  date: string;
};

type State = {
  openUpdateItemModal: boolean;
};

export type ConnectedType = {
  updateItemLoading: boolean;
  date: string;
  openUpdateItemModal: boolean;
  onOpenUpdateItem: () => void;
  onChangeDate: (date: string) => void;
  onUpdateItem: (updateItem: NewItem) => void;
  onDeleteItem: () => void;
  onCloseUpdateItem: () => void;
};

const initialState = (): State => ({
  openUpdateItemModal: false,
});

const Connected: React.FC<Props> = (props) => {
  const navigation = useNavigation();
  const [state, setState] = useState<State>(initialState());
  const setHomeDate = useSetRecoilState(homeDateState);
  const homeItems = useHomeItems();

  const { loading, data, error, refetch } = useQuery(ItemDocument, {
    variables: {
      id: props.itemID,
    },
  });

  const onOpenUpdateItem = useCallback(() => {
    setState((s) => ({ ...s, openUpdateItemModal: true }));
  }, []);

  const onCloseUpdateItem = useCallback(() => {
    setState((s) => ({ ...s, openUpdateItemModal: false }));
  }, []);

  const [updateItemMutation] = useMutation(UpdateItemDocument, {
    async onCompleted() {
      await refetch?.();

      onCloseUpdateItem();
    },
  });

  const onUpdateItem = useCallback(
    (newItem: NewItem) => {
      const updateItem: UpdateItem = {
        id: props.itemID,
        ...newItem,
      };

      const variables = {
        input: updateItem,
      };

      updateItemMutation({ variables });
    },
    [updateItemMutation, props.itemID]
  );

  const [deleteItemMutation] = useMutation(DeleteItemDocument, {
    async onCompleted() {
      await homeItems?.refetch?.();
      navigation.goBack();
    },
  });

  const onDeleteItem = useCallback(() => {
    const deleteItem: DeleteItem = {
      id: props.itemID,
    };
    const variables = {
      input: deleteItem,
    };

    deleteItemMutation({ variables });
  }, [props.itemID, deleteItemMutation]);

  const onChangeDate = useCallback(
    (date: string) => {
      const formatDate = dayjs(props.date).format('YYYY-MM-DD');
      if (formatDate !== date) {
        setHomeDate({
          date: dayjs(date).format('YYYY-MM-DDT00:00:00+09:00'),
        });

        navigation.goBack();
      }
    },
    [props, navigation, setHomeDate]
  );

  return (
    <Plain
      data={data}
      loading={loading}
      error={error}
      updateItemLoading={false}
      date={dayjs(props.date).format('YYYY-MM-DD')}
      openUpdateItemModal={state.openUpdateItemModal}
      onChangeDate={onChangeDate}
      onOpenUpdateItem={onOpenUpdateItem}
      onUpdateItem={onUpdateItem}
      onDeleteItem={onDeleteItem}
      onCloseUpdateItem={onCloseUpdateItem}
    />
  );
};

export default memo(Connected);
