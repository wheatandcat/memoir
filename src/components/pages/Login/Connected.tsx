import React, { memo, useEffect, useCallback, useState } from 'react';
import { Alert } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { storageKey, getItem } from 'lib/storage';
import TemplateLogin from 'components/templates/Login/Page';
import useFirebaseAuth from 'hooks/useFirebaseAuth';
import useHomeItems from 'hooks/useHomeItems';
import { useRecoilValue } from 'recoil';
import { authUserState } from 'store/atoms';
import {
  CreateAuthUserDocument,
  CreateAuthUserMutationVariables,
} from 'queries/api/index';
import { useNotification } from 'containers/Notification';
import { useMutation } from '@apollo/client';

type Props = {};

const Connected: React.FC<Props> = () => {
  const [loading, setLoading] = useState(false);
  const { setupAuth, onAppleLogin, onGoogleLogin, onLogout } = useFirebaseAuth(
    true,
    () => {
      setLoading(false);
    }
  );
  const { refetch } = useHomeItems();
  const { onPermissionRequest } = useNotification();
  const authUser = useRecoilValue(authUserState);
  const authenticated = !!authUser.uid;
  const navigation = useNavigation();
  const [createAuthUserMutation] = useMutation(CreateAuthUserDocument, {
    async onCompleted() {
      onPermissionRequest?.(async () => {
        await refetch?.();
        navigation.goBack();
      });
    },
    async onError() {
      // エラーになった場合はログアウトさせる
      Alert.alert('エラー', 'ログインに失敗した');
      onLogout();
    },
  });

  const createAuthUser = useCallback(async () => {
    const userID = await getItem(storageKey.USER_ID_KEY);

    const variables: CreateAuthUserMutationVariables = {
      input: {
        id: String(userID),
        isNewUser: false,
      },
    };

    createAuthUserMutation({ variables });
  }, [createAuthUserMutation]);

  useEffect(() => {
    if (authenticated) {
      createAuthUser();
    }
  }, [authenticated, createAuthUser]);

  if (!setupAuth) {
    return null;
  }

  return (
    <TemplateLogin
      loading={loading}
      onAppleLogin={async () => {
        setLoading(true);
        onAppleLogin();
      }}
      onGoogleLogin={() => {
        setLoading(true);
        onGoogleLogin();
      }}
    />
  );
};

export default memo(Connected);
