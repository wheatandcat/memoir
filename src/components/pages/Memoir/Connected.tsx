import React, { memo, useCallback, useState } from 'react';
import {
  useItemsInPeriodQuery,
  useRelationshipsQuery,
} from 'queries/api/index';
import { useNavigation } from '@react-navigation/native';
import useItemsInPeriodPaging from 'hooks/useItemsInPeriodPaging';
import { useRecoilValue } from 'recoil';
import { userState } from 'store/atoms';
import Plain from './Plain';

type Props = {
  startDate: string;
  endDate: string;
};

type State = {
  after: string | null;
  userIDList?: string[];
};

type User = {
  id: string;
  displayName: string;
  image: string;
};

export type ConnectedType = {
  startDate: string;
  endDate: string;
  users: User[];
  selectedUserIDList: string[];
  isFilter: boolean;
  onItem: () => void;
  onScreenShot: () => void;
  onLoadMore: (after: string | null) => void;
  onChangeUserID: (userIDList: string[]) => void;
};

const initialState = () => ({
  after: '',
});

const Connected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());
  const [isFilter, setIsFilter] = useState<boolean>(false);
  const user = useRecoilValue(userState);
  const navigation = useNavigation();

  const relationshipsQuery = useRelationshipsQuery({
    variables: {
      input: {
        after: '',
        first: 5,
      },
      skip: false,
    },
  });

  const queryResult = useItemsInPeriodQuery({
    variables: {
      input: {
        startDate: props.startDate,
        endDate: props.endDate,
        first: 8,
        after: state.after,
        userIDList: state.userIDList,
      },
    },
  });

  const { items, pageInfo, reset } = useItemsInPeriodPaging(queryResult, {
    merge: true,
  });

  const onLoadMore = useCallback((after: string | null) => {
    setState((s) => ({
      ...s,
      after,
    }));
  }, []);

  const onItem = useCallback(() => {}, []);

  const onChangeUserID = useCallback(
    (userIDList: string[]) => {
      setIsFilter(true);
      reset();

      setState((s) => ({ ...s, userIDList }));
    },
    [reset]
  );

  const users: User[] = [
    { id: user.userID || '', displayName: user.displayName, image: user.image },
  ];

  const relationships = relationshipsQuery.data?.relationships?.edges || [];
  const relationshipUsers: User[] = relationships.map((v) => ({
    id: v?.node?.user?.id || '',
    displayName: v?.node?.user?.displayName || '',
    image: v?.node?.user?.image || '',
  }));

  const tUsers = [...users, ...relationshipUsers];
  const selectedUserIDList = state.userIDList || tUsers.map((v) => v.id);

  const onScreenShot = useCallback(() => {
    navigation.navigate('MemoirScreenShot', {
      startDate: props.startDate,
      endDate: props.endDate,
      selectedUserIDList,
    });
  }, [props, navigation, selectedUserIDList]);

  return (
    <Plain
      startDate={props.startDate}
      endDate={props.endDate}
      isFilter={isFilter}
      items={items}
      users={tUsers}
      selectedUserIDList={selectedUserIDList}
      pageInfo={pageInfo}
      onLoadMore={onLoadMore}
      loading={queryResult.loading}
      error={queryResult.error}
      onItem={onItem}
      onScreenShot={onScreenShot}
      onChangeUserID={onChangeUserID}
    />
  );
};

export default memo(Connected);
