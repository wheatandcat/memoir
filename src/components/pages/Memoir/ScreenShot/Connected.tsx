import React, { memo } from 'react';
import {
  useItemsInPeriodQuery,
  useRelationshipsQuery,
} from 'queries/api/index';
import { User as TUser } from 'store/atoms';
import { useRecoilValue } from 'recoil';
import { userState } from 'store/atoms';
import Plain from './Plain';

export type Props = {
  startDate: string;
  endDate: string;
  selectedUserIDList: string[];
  categoryID: number;
  like: boolean;
  dislike: boolean;
};

export type ConnectedType = Omit<
  Props,
  'selectedUserIDList' | 'categoryID' | 'like' | 'dislike'
> & {
  users: User[];
};

type User = Omit<TUser, 'userID'> & {
  id: string;
};

const Connected: React.FC<Props> = (props) => {
  const user = useRecoilValue(userState);

  const relationshipsQuery = useRelationshipsQuery({
    variables: {
      input: {
        after: '',
        first: 5,
      },
      skip: false,
    },
  });

  const { data, loading, error } = useItemsInPeriodQuery({
    variables: {
      input: {
        startDate: props.startDate,
        endDate: props.endDate,
        userIDList: props.selectedUserIDList,
        categoryID: props.categoryID,
        like: props.like,
        dislike: props.dislike,
        first: 100,
        after: '',
      },
    },
  });

  const users: User[] = [
    {
      id: user.userID || '',
      displayName: user.displayName,
      image: user.image,
    },
  ];

  const relationships = relationshipsQuery.data?.relationships?.edges || [];
  const relationshipUsers: User[] = relationships.map((v) => ({
    id: v?.node?.user?.id || '',
    displayName: v?.node?.user?.displayName || '',
    image: v?.node?.user?.image || '',
  }));

  return (
    <Plain
      data={data}
      loading={loading}
      error={error}
      users={[...users, ...relationshipUsers]}
      startDate={props.startDate}
      endDate={props.endDate}
    />
  );
};

export default memo(Connected);
