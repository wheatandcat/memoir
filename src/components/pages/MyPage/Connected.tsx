import React, { memo, useCallback, useEffect } from 'react';
import { useNavigation } from '@react-navigation/native';
import TemplateMyPage from 'components/templates/MyPage/Page';
import useFirebaseAuth, { UseFirebaseAuth } from 'hooks/useFirebaseAuth';
import { useRecoilValue } from 'recoil';
import { authUserState } from 'store/atoms';
import {
  useRelationshipRequestsLazyQuery,
  useRelationshipsLazyQuery,
  RelationshipsQuery,
  useDeleteRelationshipMutation,
  useUserQuery,
} from 'queries/api/index';
import { ScreenNavigationProp as MyPageScreenNavigationProp } from './';

type Props = {};

type Relationship = NonNullable<EdgesNode<RelationshipsQuery['relationships']>>;

export type ConnectedType = {
  deleting: boolean;
  relationshipRequestCount: number;
  relationships: Relationship[];
  onRelationshipRequests: () => void;
  onUpdateProfile: () => void;
  onLogin: () => void;
  onLogout: UseFirebaseAuth['onLogout'];
  onAddShareUser: () => void;
  onDeleteRelationship: (followedId: string) => void;
};

const Connected: React.FC<Props> = () => {
  const { setup, onLogout } = useFirebaseAuth();
  const authUser = useRecoilValue(authUserState);
  const userQuery = useUserQuery();
  const navigation = useNavigation<MyPageScreenNavigationProp>();
  const [getRelationshipRequests, relationshipRequestsData] =
    useRelationshipRequestsLazyQuery({
      fetchPolicy: 'network-only',
    });
  const [getRelationships, relationshipsData] = useRelationshipsLazyQuery({
    fetchPolicy: 'network-only',
  });
  const [deleteRelationshipMutation, deleteRelationshipMutationData] =
    useDeleteRelationshipMutation({
      onCompleted() {
        relationshipsData.refetch?.();
      },
    });

  useEffect(() => {
    if (authUser.uid) {
      getRelationshipRequests({
        variables: {
          input: {
            after: '',
            first: 5,
          },
          skip: true,
        },
      });
      getRelationships({
        variables: {
          input: {
            after: '',
            first: 5,
          },
          skip: false,
        },
      });
    }
  }, [authUser.uid, getRelationshipRequests, getRelationships]);

  const onLogin = useCallback(() => {
    navigation.navigate('Login');
  }, [navigation]);

  const onUpdateProfile = useCallback(() => {
    navigation.navigate('UpdateProfile');
  }, [navigation]);

  const onAddShareUser = useCallback(() => {
    navigation.navigate('SettingAddShareUser');
  }, [navigation]);

  const onRelationshipRequests = useCallback(() => {
    navigation.navigate('SettingRelationshipRequests', {
      onCallback: () => {
        relationshipRequestsData.refetch?.();
        relationshipsData.refetch?.();
      },
    });
  }, [navigation, relationshipRequestsData, relationshipsData]);

  const onDeleteRelationship = useCallback(
    (followedId: string) => {
      deleteRelationshipMutation({
        variables: {
          followedID: followedId,
        },
      });
    },
    [deleteRelationshipMutation]
  );

  if (!setup || userQuery.loading) {
    return null;
  }

  const relationshipRequests =
    relationshipRequestsData.data?.relationshipRequests?.edges ?? [];
  const relationshipRequestCount = relationshipRequests.length;

  const relationshipEdges = relationshipsData?.data?.relationships?.edges ?? [];
  const relationships = relationshipEdges.map((v) => v.node);

  const user = {
    id: userQuery.data?.user?.id || '',
    userID: userQuery.data?.user?.id || '',
    displayName: userQuery.data?.user?.displayName || '',
    image: userQuery.data?.user?.image || '',
  };

  return (
    <TemplateMyPage
      authenticated={!!authUser.uid}
      user={user}
      relationshipRequestCount={relationshipRequestCount}
      relationships={relationships as Relationship[]}
      deleting={deleteRelationshipMutationData.loading}
      onLogout={onLogout}
      onLogin={onLogin}
      onUpdateProfile={onUpdateProfile}
      onAddShareUser={onAddShareUser}
      onRelationshipRequests={onRelationshipRequests}
      onDeleteRelationship={onDeleteRelationship}
    />
  );
};

export default memo(Connected);
