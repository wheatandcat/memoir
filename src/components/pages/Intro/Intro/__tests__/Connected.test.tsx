import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as useMemoirNotificationSetting from 'hooks/useMemoirNotificationSetting';
import { memoirNotificationSetting } from '__mockData__/memoirNotificationSetting';
import Connected, { Props } from '../Connected';

const propsData = (): Props => ({
  onFinish: jest.fn(),
});

describe('components/pages/Intro/Intro/Connected.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    jest
      .spyOn(useMemoirNotificationSetting, 'default')
      .mockImplementation((): any => ({
        ...memoirNotificationSetting(),
        loading: false,
        onSave: jest.fn(),
      }));

    wrapper = shallow(<Connected {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
