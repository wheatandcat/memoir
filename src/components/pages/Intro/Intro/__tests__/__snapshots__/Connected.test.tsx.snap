// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Intro/Intro/Connected.tsx 正常にrenderすること 1`] = `
<Memo(Plain)
  data={
    Object {
      "dayOfWeek": 1,
      "hours": 0,
      "minutes": 0,
      "notification": false,
    }
  }
  loading={false}
  onFinish={[MockFunction]}
  onSaveNotification={[Function]}
/>
`;
